src/utils/fetchWithRefresh.ts

```diff
@@ -0,0 +1,16 @@
+const fetchWithRefresh = (
```

Comment:
I feel like this function is a bit too complicated and can be simplified by moving it to the `HttpService`

Reply:
not sure, but I'll do it